clinica/iotools/converters/ukb_to_bids/ukb_utils.py

```diff
@@ -0,0 +1,369 @@
+from cmath import nan
+from os import PathLike
+from typing import BinaryIO, Iterable, List, Tuple, Union
+
+from pandas import DataFrame, notna
+
+
+def find_clinical_data(
+    clinical_data_directory: PathLike,
+) -> DataFrame:
+    from pathlib import Path
+
+    import pandas as pd
+
+    from clinica.utils.stream import cprint
+
+    # Read the xls
+    try:
+        image_data_file = list(Path(clinical_data_directory).glob("*.tsv"))
+    except StopIteration:
+        raise FileNotFoundError("Clinical data file not found.")
+
+    # Assign the dfs
+    df_list = []
+    for i in range(0, len(image_data_file)):
+        df_list.append(pd.read_csv(str(image_data_file[i]), sep="\t"))
+        df_list[i] = df_list[i].set_index(df_list[i].axes[1][0])
+        if df_list[i].index.name == "eid":
+            df_clinical = df_list[i]
+
+    # Warn if a dataframe is missing
+    if "df_clinical" in locals():
```

Comment:
It is usually better to avoid this kind of pattern if possible. I would simply initialize df_clinical to None before the for loop and test here whether it was assigned or not.

```python
if df_clinical is not None:
    cprint("bla bla")
else:
    raise("bla bla")
```